src/Microsoft.IdentityModel.Tokens/Validation/Results/Details/AudienceValidationError.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+
+#nullable enable
+namespace Microsoft.IdentityModel.Tokens
+{
+    internal class AudienceValidationError : ValidationError
+    {
+        private string? _invalidAudience;
+
+        public AudienceValidationError(
+            MessageDetail messageDetail,
+            Type exceptionType,
+            StackFrame stackFrame,
+            IList<string>? invalidAudiences)
+            : base(messageDetail, ValidationFailureType.AudienceValidationFailed, exceptionType, stackFrame)
+        {
+            _invalidAudience = Utility.SerializeAsSingleCommaDelimitedString(invalidAudiences);
```

Comment:
Do we need to do this in the ctor?
There is a possibility that this string will never be used.

Reply:
Moved out of the ctor.